pkg/factory/risingwave_object_factory.go

```diff
@@ -418,7 +418,7 @@ func (f *RisingWaveObjectFactory) envsForEtcd() []corev1.EnvVar {
 			ValueFrom: &corev1.EnvVarSource{
 				SecretKeyRef: &corev1.SecretKeySelector{
 					LocalObjectReference: secretRef,
-					Key:                  consts.SecretKeyEtcdUsername,
+					Key:                  *credentials.UsernameKeyRef,
```

Comment:
I suddenly realized that `string` might be better than `*string` in such cases. Shall we just use strings for all key references?